From a480a4223639766f4f0625434a28204b014fd882 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Tue, 15 Jul 2025 10:31:08 +0200
Subject: [PATCH] pyproject.toml: remove it in favor of setup.py

This patch is I would say kinda hacky. OpenWrt
can compile Python packages with pyproject.toml,
but this pyproject.toml requires the latest
setuptools, which is shipped separately these days.

But Python 3.11, which is used in OpenWrt
is using bundled setuptools together with pip
and the version of setuptools does not support
ext-modules [1].

[1] https://github.com/pypa/setuptools/pull/4568

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 pyproject.toml | 37 -------------------------------------
 1 file changed, 37 deletions(-)
 delete mode 100644 pyproject.toml

--- a/pyproject.toml
+++ /dev/null
@@ -1,37 +0,0 @@
-[build-system]
-requires = ["setuptools", "wheel"]
-build-backend = "setuptools.build_meta"
-
-[project]
-name = "pyuci"
-dynamic = []
-description = "Python UCI bindings"
-readme = "README.md"
-license = {file = "LICENSE"}
-version = "0.10.3"
-requires-python = ">=3.7"
-authors = [
-    { name = "CZ.NIC z.s.p.o", email = "karel.koci@nic.cz" },
-]
-classifiers = [
-    "Development Status :: 5 - Production/Stable",
-    "License :: OSI Approved :: MIT License",
-    "Operating System :: POSIX :: Linux",
-    "Programming Language :: Python :: 3",
-]
-
-[project.urls]
-Homepage = "https://gitlab.nic.cz/turris/pyuci"
-
-[tool.setuptools]
-packages = ["euci"]
-
-[[tool.setuptools.ext-modules]]
-name = "uci"
-sources = ["ucimodule.c", "pyuci.c", "pyhelper.c"]
-language = "c"
-libraries = ["uci"]
-
-# Uncomment to enable coverage measurement
-#extra-compile-args = ["-fprofile-arcs", "-ftest-coverage"]
-#extra-link-args = ["-fprofile-arcs"]
